packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2601,17 +2610,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         serializedContent: string,
         serverMaxOpSize: number,
         batch: boolean,
-        opMetadataInternal: unknown = undefined,
+        opMetadataInternal: Record<string, unknown> | undefined = undefined,
     ): number {
+        let maybeCompressedContent: any;
+        let metadata: Record<string, unknown> | undefined;
+        let contentLength: number;
+
+        if (this.runtimeOptions.compressionOptions?.minimumSize !== undefined &&
+            serializedContent &&
+            serializedContent.length > this.runtimeOptions.compressionOptions.minimumSize
+        ) {
+            this.compressedOpCount++;
+
+            const compressionStart = Date.now();
+            const contentsAsBuffer = new TextEncoder().encode(serializedContent);
+            const compressedContents = compress(contentsAsBuffer);
+            const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
+            const duration = Date.now() - compressionStart;
+
+            if (this.compressedOpCount % 100) {
+                this.mc.logger.sendPerformanceEvent({
+                    eventName: "compressedOp",
+                    duration,
+                    sizeBeforeCompression: serializedContent.length,
+                    sizeAfterCompression: compressedContent.length,
+                });
+            }
+
+            if (serializedContent.length < compressedContent.length) {
+                maybeCompressedContent = content;
+                metadata = opMetadataInternal;
```

Comment:
I am assuming here that the omission of `compressed: true` is the same as it being false, just wanted to point of if that's not the case.

Reply:
Yep, no compression property is equivalent to it being false